paths.go

```diff
@@ -0,0 +1,128 @@
+package quamina
+
+import (
+	"fmt"
+	"strings"
+)
+
+const PathSeparator = "\n"
+
+type pathsIndex struct {
```

Comment:
If this implements HierarchicalNameTracker then I think it's polite to have a comment that says so. I'm mostly not 100% a fan of Go's silent interface implementation.

Reply:
> I'm mostly not 100% a fan of Go's silent interface implementation.

I always had a conflicted feelings coming from C# to Go about this, and will add a comment to help with those feelings :)

Reply:
Added comments.